File: 413-arithmetic-slices/413-arithmetic-slices.cpp
class Solution {
public:
    int numberOfArithmeticSlices(vector<int>& nums) {
        int res = 0, curr = 0;
        int n = (int) nums.size();
        for(int i = 1; i < n - 1; i++) {
            if(nums[i] - nums[i - 1] == nums[i + 1] - nums[i]) {
                curr++;
                res += curr;
            } else {
                curr = 0;
            }
        }
        
        return res;
    }
};

File: 413-arithmetic-slices/413-arithmetic-slices.java
class Solution {
    public int numberOfArithmeticSlices(int[] nums) {
        int n = nums.length;
        int dp = 0;
        int res = 0;
        
        for(int i = 1; i < n - 1; i++) {
            int prev = nums[i - 1];
            int curr = nums[i];
            int next = nums[i + 1];
            
            if(curr - prev == next - curr) {
                dp += 1;
                res += dp;
            } else {
                dp = 0;
            }
        }
        
        return res;
    }
}